tangelo/algorithms/projective/tests/test_qpe.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2023 Good Chemsitry Company.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+import numpy as np
+from openfermion import get_sparse_operator
+
+from tangelo.algorithms.projective.qpe import QPESolver
+from tangelo.toolboxes.ansatz_generator.ansatz_utils import trotterize
+from tangelo.toolboxes.operators import QubitOperator
+from tangelo.toolboxes.qubit_mappings.mapping_transform import fermion_to_qubit_mapping
+from tangelo.toolboxes.qubit_mappings.statevector_mapping import get_reference_circuit
+from tangelo.molecule_library import mol_H2_sto3g
+from tangelo.linq.helpers.circuits.statevector import StateVector
+
+
+class QPESolverTest(unittest.TestCase):
+
+    def test_instantiation(self):
+        """Try instantiating QPESolver with basic input."""
+
+        options = {"molecule": mol_H2_sto3g, "qubit_mapping": "jw"}
+        QPESolver(options)
+
+    def test_instantiation_incorrect_keyword(self):
+        """Instantiating with an incorrect keyword should return an error """
+
+        options = {"molecule": mol_H2_sto3g, "qubit_mapping": "jw", "dummy": True}
+        self.assertRaises(KeyError, QPESolver, options)
+
+    def test_instantiation_missing_molecule(self):
+        """Instantiating with no molecule should return an error."""
+
+        options = {"qubit_mapping": "jw"}
+        self.assertRaises(ValueError, QPESolver, options)
+
+    def test_simulate_h2(self):
+        """Run QPE on H2 molecule, with JW qubit mapping and exact simulator
+        """
+
+        qpe_options = {"molecule": mol_H2_sto3g, "qubit_mapping": "scbk", "up_then_down": True, "size_qpe_register": 8,
+                       "backend_options": {"target": "qulacs"}, "unitary_options": {"time": 2*np.pi, "n_trotter_steps": 1,
+                                                                                    "n_steps_method": "repeat", "trotter_order": 4}}
+        qpe_solver = QPESolver(qpe_options)
+        qpe_solver.build()
+
+        energy = qpe_solver.simulate()
+        self.assertAlmostEqual(energy, 0.13671875, delta=1e-4)
+
+        # Test that get_resources returns expected results
```

Comment:
A comment here: In other algorithms (VQE), some of these numbers were implementation-dependent or relied on arbitrary things (ex: order in which qubit Hamiltonian terms were visited to produce the ansatz, final value of var params after the optimization, ...).

These numbers to me are not all a symbol of hard truth (maybe a simple circuit optimization pass done inside the algorithm would change them, into an equivalent circuit that is just as correct), so I'm unsure about testing all of them and causing the test to break.

Maybe we could settle down for only testing circuit width and number of terms ?
As for the two others, we now know that they are numbers that seems to make sense and are correctly assigned but I don't think we should make them a requirement.